crates/re_ui/examples/re_ui_example.rs

```diff
@@ -600,3 +645,788 @@ impl egui_tiles::Behavior<Tab> for MyTileTreeBehavior {
         }
     }
 }
+
+// ==============================================================================
+// DRAG AND DROP DEMO
+
+mod drag_and_drop {
+    use std::collections::HashSet;
+
+    #[derive(Hash, Clone, Copy, PartialEq, Eq)]
+    struct ItemId(u32);
+
+    pub struct ExampleDragAndDrop {
+        items: Vec<(ItemId, String)>,
```

Comment:
This is confusing.

This lets me swap the `String` of two rows in the `items` list, and leave their `ItemId` in place, creating a mismatch between `ItemId` and the `String`. Is this intentional? As a reader, I don't know yet. I'm suspecting the answer is "no", which is why I'm writing this and suggest you implement `Display` on `ItemId` instead.

Make invalid states impossible!

---

However, if it is indeed intentional, this field should have a docstring explaining itself. Is this an ordered lookup map? Should this be an [indexmap](https://docs.rs/indexmap/latest/indexmap/) instead?

Is this a valid configuration? What does it mean?

```
[
    (`2`, "Item 0"),
    (`0`, "Item 1"),
    (`0`, "Item 5"),
    (`1`, "Lol"),
]
```


The type system is a powerful way to communicate with the reader

Reply:
I removed that string altogether and now derive the label from the `ItemId`. It's not relevant for that demo anyways.